src/main/java/seedu/duke/ui/AppUi.java

```diff
@@ -0,0 +1,287 @@
+     * and description (if provided). If no new value is provided for amount or description, the
+     * existing values are retained. Finally, an {@link EditEntryCommand} is created and executed
+     * to apply the changes to the financial list.
+     *
+     * @param commandArguments A map of parsed command arguments that contains the entry index and
+     *                         optional new values for the amount ("/a") and description ("/des").
+     */
+    public void editEntry(HashMap<String, String> commandArguments) {
+        int index = Integer.parseInt(commandArguments.get("argument"));
+
+        assert index > 0 : "Index of entry to edit must be greater than 0";
+        assert index <= financialList.getEntryCount() : "Index of entry to edit must be within the list size";
+
+        FinancialEntry entry = financialList.getEntry(index - 1);
+
+        String amountStr = commandArguments.get("/a");
+        double amount = (amountStr != null) ? Double.parseDouble(amountStr) : entry.getAmount();
+
+        String description = commandArguments.getOrDefault("/des", entry.getDescription());
+
+        EditEntryCommand editEntryCommand = new EditEntryCommand(index, amount, description);
+        editEntryCommand.execute(financialList);
+    }
+
+    /**
+     * Deletes an existing entry from the financial list based on the provided command arguments.
+     *
+     * The method extracts the index of the entry to be deleted from the command arguments.
+     * A {@link DeleteCommand} is created and executed to remove the entry from the financial list.
+     *
+     * @param commandArguments A map of parsed command arguments that contains the index of the entry
+     *                         to be deleted.
+     */
+    public void deleteEntry(HashMap<String, String> commandArguments) {
+        int index = Integer.parseInt(commandArguments.get("argument"));
+
+        DeleteCommand deleteCommand = new DeleteCommand(index);
+        deleteCommand.execute(financialList);
+    }
+
+    /**
+     * This method helps execute the appropriate command based on the "argument"
+     * provided in the commandArguments HashMap. If the argument is "expense", it will
+     * execute the SeeAllExpensesCommand. If the argument is "income", it will execute
+     * the SeeAllIncomesCommand. If no argument or an unknown argument is provided,
+     * it defaults to executing SeeAllEntriesCommand to list all entries.
+     *
+     * @param commandArguments A HashMap containing the command argument with the key "argument".
+     *                         The value can be "expense", "income", or null/empty for listing all entries.
+     */
+    public void listHelper(HashMap<String, String> commandArguments) throws FinanceBuddyException {
+        String type = commandArguments.get("argument");
+        String start = commandArguments.get("/from");
+        String end = commandArguments.get("/to");
```

Comment:
Maybe you can better name these variables to `startDateString`, as it might be confusing alongside `startDate` and `endDate`.